### PR TITLE
Prevent editing, deleting and assigning groups/users to system roles

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [X] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.roles.v2/components/edit-role/edit-role.tsx [hunk 1]
```diff
@@ -81,9 +81,10 @@ export const EditRole: FunctionComponent<EditRoleProps> = (props: EditRoleProps)
 
     const isReadOnly: boolean = useMemo(() => {
         return !isFeatureEnabled(featureConfig,
-            LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE")) ||
-            !hasRequiredScopes(featureConfig,
-                featureConfig?.scopes?.update, allowedScopes);
+            LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE"))
+            || !hasRequiredScopes(featureConfig,
+                featureConfig?.scopes?.update, allowedScopes)
```
Comment (Achintha444) [inline features/admin.roles.v2/components/edit-role/edit-role.tsx:86]: add organization type as well

File: features/admin.roles.v2/components/role-list.tsx [hunk 2]
```diff
@@ -292,14 +293,19 @@ export const RoleList: React.FunctionComponent<RoleListProps> = (props: RoleList
...
+                        role?.displayName === CommonRoleConstants.ADMIN_ROLE ||
                         role?.displayName === CommonRoleConstants.ADMIN_GROUP ||
-                        role?.displayName === administratorRoleDisplayName)
+                        role?.displayName === administratorRoleDisplayName
+                    )
                     || !isFeatureEnabled(userRolesFeatureConfig,
                         RoleConstants.FEATURE_DICTIONARY.get("ROLE_DELETE"))
                     || !hasRequiredScopes(userRolesFeatureConfig,
-                        userRolesFeatureConfig?.scopes?.delete, allowedScopes),
+                        userRolesFeatureConfig?.scopes?.delete, allowedScopes);
```
Comment (Achintha444) [inline features/admin.roles.v2/components/role-list.tsx:307]: add organization type as well

File: features/admin.roles.v2/components/edit-role/edit-role.tsx (hunk 1 above)
Comment (pavinduLakshan) [reply features/admin.roles.v2/components/edit-role/edit-role.tsx:86]: No need to add organization type as system roles are already handled as readonly by `systemRole` property.

File: features/admin.roles.v2/components/role-list.tsx (hunk 2 above)
Comment (pavinduLakshan) [reply features/admin.roles.v2/components/role-list.tsx:307]: No need to add organization type as system roles are already handled as readonly by `systemRole` property.

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**